packages/api-fetch/src/middlewares/preloading.js

```diff
@@ -68,15 +68,39 @@ function createPreloadingMiddleware( preloadedData ) {
  * @return {Promise<any>} Promise with the response.
  */
 function prepareResponse( responseData, parse ) {
-	return Promise.resolve(
-		parse
-			? responseData.body
-			: new window.Response( JSON.stringify( responseData.body ), {
-					status: 200,
-					statusText: 'OK',
-					headers: responseData.headers,
-			  } )
-	);
+	if ( parse ) {
+		return Promise.resolve( responseData.body );
+	}
+
+	try {
+		return Promise.resolve(
+			new window.Response( JSON.stringify( responseData.body ), {
+				status: 200,
+				statusText: 'OK',
+				headers: responseData.headers,
+			} )
+		);
+	} catch {
+		Object.entries( responseData.headers ).forEach( ( [ key, value ] ) => {
```

Comment:
```suggestion
		// See: https://github.com/WordPress/gutenberg/issues/67358#issuecomment-2621163926.
		Object.entries( responseData.headers ).forEach( ( [ key, value ] ) => {
```

Let's add a note for future reference.

Reply:
good idea, done @Mamaduka  ✅